programs/fileio.c

```diff
@@ -1677,6 +1677,29 @@ int FIO_compressMultipleFilenames(FIO_prefs_t* const prefs,
     /* init */
     assert(outFileName != NULL || suffix != NULL);
     if (outFileName != NULL) {   /* output into a single destination (stdout typically) */
+        if (nbFiles > 1) {
+            if (!strcmp (outFileName, stdoutmark)) {
+                DISPLAY("zstd: WARNING: all input files will be processed and concatenated into stdout. ");
+            } else {
+                DISPLAY("zstd: WARNING: all input files will be processed and concatenated into a single output file: %s ", outFileName);
+            }
+            if (prefs->removeSrcFile && !prefs->overwrite) {
+                DISPLAY("\nYou must specify -f as well in order to execute this command with --rm. Aborting...");
+                return 1;
+            }
+            
+            DISPLAY("Proceed? (y/n): ");
```

Comment:
I think the behavior we want is:
* If `-f` is passed then proceed without waiting for confirmation. This should probably also silence the warnings.
* If `-f` is not passed and `--rm` is not passed then print a warning and proceed. This is a non-destructive operation, but maybe not what they want.
* If `-f` is not passed and `--rm` is passed then print a warning. At this point we could either abort, or ask for confirmation. Asking for confirmation is good for me.

Reply:
minor : 
there is already a `(y/n)` prompt at line 602,
it might be ground to unify both implementations into a single function.

Reply:
yep i was just working on this!

Reply:
Prompting the user (i.e., blocking the process for a response) makes me a little nervous. At the very least, we should gate such a prompt behind `isatty(STDIN_FILENO)`.

Reply:
We already have this situation regarding confirmation to overwrite existing files.

Things we currently do to make such a scenario compatible with scripts (which can't answer prompt) : 
`-f` : is an automatic "yes"
`-q` : is an automatic "no"
`-f -q` : `-f` prevails

Reply:
Oh interesting - wasn't aware of that flag. I'll add that in.

Reply:
Right. But it would be very easy, without those flags, to get into this situation in a non-interactive environment, and then `zstd` would just hang.

Reply:
I believe that's what already happens now when doing `zstd FILE` and `FILE.zst` already exist.
`zstd FILE` asks for confirmation prompt.
But `zstd FILE < /dev/null`, which is similar to non-interactive, is an automatic "no".
I presume it's a natural consequence of `getchar()`

I tested the behavior using `gzip` and it's exactly the same.

Reply:
Hmmmm, then I think this is actually an unlikely, but potentially bad, bug:

```
$ cat <<EOF >test
> y
> Hello, world!
> EOF
$ ./zstd test
test                 :181.25%   (    16 =>     29 bytes, test.zst)             
$ ./zstd -q -d test.zst -c
y
Hello, world!
$ ./zstd < test -o test.zst
/*stdin*\            :192.86%   (    14 =>     27 bytes, test.zst)             
$ ./zstd -q -d test.zst -c
Hello, world!
$ 
```

I.e., the prompt can end up swallowing part of the input when it's being presented via `stdin`.

Reply:
Indeed, there should not be any usage of `getchar()` when any input is `stdin`,
and therefore it should an automatic "no".

That's a bug to fix.

Unsure if it's "fair" to make it part of _this_ PR, as it increases its scope.

Reply:
Yeah, makes sense. I'll put up a PR after this gets merged, since this is refactoring that code already.

Reply:
Good point !

I think the initial intention is that a level `2` enables live line status, 
such as `Read : 310 / 953 MB ==> 35%`, 
while level `1` quiets that, and only prints warnings.

It was deemed that line status was designed for a single file or stream, 
and as a consequence was too annoying for multiple files, 
due to a combination of confusion and space occupation.

On short term, the "easy" way out is that
processing multiple files is by default equivalent to `-q` , 
meaning it effectively disables prompt selection.

Note that prompt selection still remains a thing with an explicit `-v`.

Longer term, we may have to consider if we want a concise line status dedicated to multiple files,
so that level `2` can still exist in combination with multiple files.